### ISSUE TITLE
Smoking Racks can no longer be used as a tool in crafting recipes

### ISSUE THREAD

**Issue by chemicalhorse** [CleverRaven/Cataclysm-DDA#52755]
### Describe the bug

This is despite smoking racks still being listed as a usable tool in recipes.

### Steps To Reproduce

1. Build a Smoking Rack
2. Fill smoking rack with charcoal
3. Attempt to craft some food (say clean water) 
4. Be unable to do so

### Expected behavior

Either smoking racks being able to be used as a tool again, or the removal of smoking racks being listed as a usable tool in recipes.

### Screenshots

![smoke1](https://user-images.githubusercontent.com/93713285/141248353-56347124-38c2-43e2-acc8-e9f299ed7795.png)
![smoke2](https://user-images.githubusercontent.com/93713285/141248356-09d55bb3-ca8c-4a2e-90d0-96769929c1ca.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: e79ee6a [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Extra Mutated Scenarios [extra_mut_scens],
    SpeedyDex [speedydex],
    Stats Through Skills [StatsThroughSkills],
    Stats Through Kills [stats_through_kills]
]

**Comment by PatrikLundell** [comment]
I believe this should have been fixed by #52721 (sorry for screwing it up with #52674), but that was merged earlier than the version referenced.

I tried to repeat the problem locally using the branch that provided #52721 and failed to repeat it. I first constructed a smoking rack from the build menu and then spawned meat, water, and charcoal. I then loaded the charcoal into the rack and brought up the crafting menu. the meat could be cooked using "basecamp charcoal smoker" in green, and water couldn't be purified due to the lack of a pot, but again the smoker was available as a heat source.

I then move away a bit and spawned a smoking rack item (which was spawned filled with charcoal). Again, cooking meat was available, but this time with "smoking rack" as the available heat source, and likewise, pure water crafting had it available as a heat source for its crafting.

**Comment by chemicalhorse** [comment]
I just updated to most recent experimental and I'm afraid I still seem to be experiencing the issue

**Comment by PatrikLundell** [comment]
I just retried it with the latest experimental as of a couple of minutes ago (just spawning meat and the smoking rack item) and it works there for me. I'm using an old save (I think it's from 0.E2, but might be 0.E experimental) that uses only the default "mods". I'm on Windows as well (although it's an updating 10.1 version).

When looking at SPAM I get "smoking rack" as the top item listed, but I suspect the list is unordered, with the order jumping around from game session to game session, so that's probably not relevant.

I AM able to replicate your situation (with the smoking rack far down the list as well) if I move quite some distance away from the smoking rack (I'm 7 steps away) so the rack isn't within the tool usage range, but I assume that's not your case.

**Comment by Setokaiva** [comment]
In version 74156e1 I thought I had this issue too, until I realized that the _smoking rack_ literally WAS being treated as a _charcoal smoker_ in crafting recipes, just like its description says. Mind you, I'm using the constructed full-size one. Rather than show a smoking rack with charges, recipes showed the charcoal smoker, despite me not actually having a charcoal smoker in my base -- believe me, I looked.
